### PR TITLE
Move cipher note from kerb to tls page

### PR DESCRIPTION
**Pull request by Ordinant**

### CODE REVIEW

File: docs/src/main/sphinx/security/kerberos.rst [hunk 1]
```diff
@@ -128,25 +128,8 @@ Property                                                  Description
...
-
-    .. code-block:: text
-
-        http-server.https.included-cipher=TLS_RSA_WITH_AES_128_CBC_SHA,TLS_RSA_WITH_AES_128_CBC_SHA256
-        http-server.https.excluded-cipher=
-
-    The Java documentation lists the `supported cipher suites
-    <https://docs.oracle.com/en/java/javase/11/security/oracle-providers.html#GUID-7093246A-31A3-4304-AC5F-5FB6400405E2__SUNJSSE_CIPHER_SUITES>`_.
+See :ref:`Standards supported <tls-version-and-ciphers>` for a discussion of the
+TLS versions and cipher suites supported by the Trino coordinator.
```
[the remark this answers is not in the document]
Comment (Ordinant) [reply docs/src/main/sphinx/security/kerberos.rst:132]: ✅

File: docs/src/main/sphinx/security/tls.rst [hunk 2]
```diff
@@ -12,17 +12,54 @@ This topic describes how to configure your Trino server to use :ref:`TLS
...
 .. important::
 
     This page discusses only how to prepare the Trino server for secure client
     connections from outside of the Trino cluster to its coordinator.
 
 See the :doc:`Glossary </appendix/glossary>` to clarify unfamiliar terms.
 
+.. _tls-version-and-ciphers:
+
+Standards supported
```
[the remark this answers is not in the document]
Comment (Ordinant) [reply docs/src/main/sphinx/security/tls.rst:24]: ✅

File: docs/src/main/sphinx/security/tls.rst [hunk 3]
```diff
@@ -12,17 +12,54 @@ This topic describes how to configure your Trino server to use :ref:`TLS
...
+-------------------
+
+When configured to use TLS, the Trino server responds to client connections
+using TLS 1.2 and TLS 1.3 certificates. The server rejects TLS 1.1, TLS 1.0, and
+all SSL format certificates.
+
+The Trino server does not specify a set of supported ciphers, instead deferring
+to the JVM's defaults. The Java documentation lists the `supported cipher suites
+<https://docs.oracle.com/en/java/javase/11/security/oracle-providers.html#GUID-7093246A-31A3-4304-AC5F-5FB6400405E2__SUNJSSE_CIPHER_SUITES>`_.
```
[the remark this answers is not in the document]
Comment (Ordinant) [reply docs/src/main/sphinx/security/tls.rst:33]: added same JVM from the same vendor as configured on the coordinator.

The exports issue comes up if you're specifying your own cipher list: you may have to download a different cipher suite package. Note added later on the page.

File: docs/src/main/sphinx/security/tls.rst [hunk 4]
```diff
@@ -12,17 +12,54 @@ This topic describes how to configure your Trino server to use :ref:`TLS
...
+
+Run the following single-line code to determine the JVM's default cipher list.
```
[the remark this answers is not in the document]
Comment (Ordinant) [reply docs/src/main/sphinx/security/tls.rst:35]: done

File: docs/src/main/sphinx/security/tls.rst [hunk 5]
```diff
@@ -12,17 +12,54 @@ This topic describes how to configure your Trino server to use :ref:`TLS
...
+.. code-block:: java
+
+  jrunscript -e "java.util.Arrays.asList(javax.net.ssl.SSLServerSocketFactory.getDefault().getSupportedCipherSuites()).stream().forEach(println)"
+
+The default Trino server specifies a set of regular expressions that exclude
+older cipher suites that do not support forward secrecy (FS).
+
+Use the ``http-server.https.included-cipher`` property to specify a
+comma-separated list of ciphers in preferred use order. If one of your preferred
+selections is a non FS cipher, you must also set the
```
[the remark this answers is not in the document]
Comment (Ordinant) [reply docs/src/main/sphinx/security/tls.rst:46]: ✅

File: docs/src/main/sphinx/security/tls.rst [hunk 6]
```diff
@@ -12,17 +12,54 @@ This topic describes how to configure your Trino server to use :ref:`TLS
...
+default exclusions. For example:
+
+.. code-block:: text
+
+   http-server.https.included-cipher=TLS_RSA_WITH_AES_128_CBC_SHA,TLS_RSA_WITH_AES_128_CBC_SHA256
+   http-server.https.excluded-cipher=
+
+.. note::
+
+    Monitor the CPU usage on the Trino coordinator after enabling HTTPS. Java
```
[the remark this answers is not in the document]
Comment (Ordinant) [reply docs/src/main/sphinx/security/tls.rst:57]: added this to the Note.

**Review by mosabua — APPROVED** [review]

File: docs/src/main/sphinx/security/tls.rst [hunk 7]
```diff
@@ -12,17 +12,67 @@ This topic describes how to configure your Trino server to use :ref:`TLS
...
+to the defaults set by the JVM version in use. The documentation for Java 11
+lists its `supported cipher suites
+<https://docs.oracle.com/en/java/javase/11/security/oracle-providers.html#GUID-7093246A-31A3-4304-AC5F-5FB6400405E2__SUNJSSE_CIPHER_SUITES>`_.
+
+Run the following single-line code on the same JVM from the same vendor as
+configured on the coordinator to determine that JVM's default cipher list.
+
+.. code-block:: java
+
+  jrunscript -e "java.util.Arrays.asList(javax.net.ssl.SSLServerSocketFactory.getDefault().getSupportedCipherSuites()).stream().forEach(println)"
```
Comment (electrum) [inline docs/src/main/sphinx/security/tls.rst:41]: This doesn't work on Java 17 (with Zulu JDK on macOS):
```
script engine for language js can not be found
```
This should work on both:
```shell
echo "java.util.Arrays.asList(((javax.net.ssl.SSLServerSocketFactory) javax.net.ssl.SSLServerSocketFactory.getDefault()).getSupportedCipherSuites()).stream().forEach(System.out::println)" | jshell -

Comment (Ordinant) [reply docs/src/main/sphinx/security/tls.rst:41]: OK. Fixing to use the better script. Thanks!

File: docs/src/main/sphinx/security/tls.rst [hunk 8]
```diff
@@ -12,17 +12,68 @@ This topic describes how to configure your Trino server to use :ref:`TLS
...
+
+The Trino server does not specify a set of supported ciphers, instead deferring
+to the defaults set by the JVM version in use. The documentation for Java 11
+lists its `supported cipher suites
+<https://docs.oracle.com/en/java/javase/11/security/oracle-providers.html#GUID-7093246A-31A3-4304-AC5F-5FB6400405E2__SUNJSSE_CIPHER_SUITES>`_.
+
+Run the following two-line code on the same JVM from the same vendor as
+configured on the coordinator to determine that JVM's default cipher list.
+
+.. code-block:: java
```
Comment (electrum) [inline docs/src/main/sphinx/security/tls.rst:39]: This should be `shell`

Comment (Ordinant) [reply docs/src/main/sphinx/security/tls.rst:39]: ✅

**Comment by mosabua** [comment]
@dain @electrum @hashhar can we move this forward?

**Review by dain — COMMENTED** [review]
looks good to me

**Comment by Ordinant** [comment]
Dain approved! Who's going to merge this in, please?

File: docs/src/main/sphinx/security/tls.rst [hunk 9]
```diff
@@ -12,17 +12,68 @@ This topic describes how to configure your Trino server to use :ref:`TLS
...
 .. important::
 
     This page discusses only how to prepare the Trino server for secure client
     connections from outside of the Trino cluster to its coordinator.
 
 See the :doc:`Glossary </appendix/glossary>` to clarify unfamiliar terms.
 
+.. _tls-version-and-ciphers:
+
+Supported standards
```
Comment (hashhar) [inline docs/src/main/sphinx/security/tls.rst:24]: Very weird title IMO.

"Supported TLS versions and Cipher suites" is more factual and actually conveys information.

File: docs/src/main/sphinx/security/tls.rst [hunk 10]
```diff
@@ -12,17 +12,68 @@ This topic describes how to configure your Trino server to use :ref:`TLS
...
+
+Run the following two-line code on the same JVM from the same vendor as
+configured on the coordinator to determine that JVM's default cipher list.
+
+.. code-block:: shell
+
+  echo "java.util.Arrays.asList(((javax.net.ssl.SSLServerSocketFactory) \
+  javax.net.ssl.SSLServerSocketFactory.getDefault()).getSupportedCipherSuites()).stream().forEach(System.out::println)" | jshell -
+
+The default Trino server specifies a set of regular expressions that exclude
```
Comment (hashhar) [inline docs/src/main/sphinx/security/tls.rst:44]:
```suggestion
By default Trino is configured to exclude
```

File: docs/src/main/sphinx/security/tls.rst [hunk 11]
```diff
@@ -12,17 +12,68 @@ This topic describes how to configure your Trino server to use :ref:`TLS
...
+.. _tls-version-and-ciphers:
+
+Supported standards
+-------------------
+
+When configured to use TLS, the Trino server responds to client connections
+using TLS 1.2 and TLS 1.3 certificates. The server rejects TLS 1.1, TLS 1.0, and
+all SSL format certificates.
+
+The Trino server does not specify a set of supported ciphers, instead deferring
```
Comment (hashhar) [inline docs/src/main/sphinx/security/tls.rst:31]: This statement and the next paragraph are conflicting. Either Trino can exclude and include ciphers or it cannot.

File: docs/src/main/sphinx/security/tls.rst [hunk 12]
```diff
@@ -12,17 +12,68 @@ This topic describes how to configure your Trino server to use :ref:`TLS
...
+to the defaults set by the JVM version in use. The documentation for Java 11
+lists its `supported cipher suites
```
Comment (hashhar) [inline docs/src/main/sphinx/security/tls.rst:33]:
```suggestion
to the defaults set by the JVM version in use. For example Oracle JDK 11
lists its `supported cipher suites
```

File: docs/src/main/sphinx/security/tls.rst [hunk 13]
```diff
@@ -12,17 +12,68 @@ This topic describes how to configure your Trino server to use :ref:`TLS
...
+<https://docs.oracle.com/en/java/javase/11/security/oracle-providers.html#GUID-7093246A-31A3-4304-AC5F-5FB6400405E2__SUNJSSE_CIPHER_SUITES>`_.
+
+Run the following two-line code on the same JVM from the same vendor as
+configured on the coordinator to determine that JVM's default cipher list.
```
Comment (hashhar) [inline docs/src/main/sphinx/security/tls.rst:37]:
```suggestion
Run the following code on the same JVM as used by
the coordinator to determine that JVM's default cipher list.
```

File: docs/src/main/sphinx/security/tls.rst (hunk 10 above)
Comment (hashhar) [reply docs/src/main/sphinx/security/tls.rst:44]: The configs don't use regex at all. It's just a list.

File: docs/src/main/sphinx/security/tls.rst [hunk 14]
```diff
@@ -12,17 +12,68 @@ This topic describes how to configure your Trino server to use :ref:`TLS
...
+.. code-block:: shell
+
+  echo "java.util.Arrays.asList(((javax.net.ssl.SSLServerSocketFactory) \
+  javax.net.ssl.SSLServerSocketFactory.getDefault()).getSupportedCipherSuites()).stream().forEach(System.out::println)" | jshell -
+
+The default Trino server specifies a set of regular expressions that exclude
+older cipher suites that do not support forward secrecy (FS).
+
+Use the ``http-server.https.included-cipher`` property to specify a
+comma-separated list of ciphers in preferred use order. If one of your preferred
```
Comment (hashhar) [inline docs/src/main/sphinx/security/tls.rst:48]:
```suggestion
comma-separated list of ciphers in order of preferred use. If one of your preferred
```

File: docs/src/main/sphinx/security/tls.rst [hunk 15]
```diff
@@ -12,17 +12,68 @@ This topic describes how to configure your Trino server to use :ref:`TLS
...
+considered in conjunction with your organization's security managers. Using a
+different suite may require downloading and installing a different SunJCE
+implementation package. Some locales may have export restrictions on cipher
+suites. See the discussion in Java documentation that begins with `Customizing
+the Encryption Algorithm Providers
+<https://docs.oracle.com/en/java/javase/11/security/java-secure-socket-extension-jsse-reference-guide.html#GUID-316FB978-7588-442E-B829-B4973DB3B584>`_.
+
+.. note::
+
+    If you manage the coordinator's direct TLS implementatation, monitor the CPU
```
Comment (hashhar) [inline docs/src/main/sphinx/security/tls.rst:68]: What a "direct TLS implementation"?

File: docs/src/main/sphinx/security/tls.rst [hunk 16]
```diff
@@ -12,17 +12,68 @@ This topic describes how to configure your Trino server to use :ref:`TLS
...
+    usage on the Trino coordinator after enabling HTTPS. Java prefers the more
+    CPU-intensive cipher suites, if you allow it to choose from a big list of
```
Comment (hashhar) [inline docs/src/main/sphinx/security/tls.rst:70]: this comma is redundant

File: docs/src/main/sphinx/security/tls.rst [hunk 17]
```diff
@@ -12,17 +12,68 @@ This topic describes how to configure your Trino server to use :ref:`TLS
...
+    ciphers. If the CPU usage is unacceptably high after enabling HTTPS, you can
+    configure Java to use specific cipher suites as described in this section.
```
Comment (hashhar) [inline docs/src/main/sphinx/security/tls.rst:72]: Can be read to mean to verbatim set the config shared above which I don't beleive is what we intend to mean here.

File: docs/src/main/sphinx/security/tls.rst [hunk 18]
```diff
@@ -12,17 +12,68 @@ This topic describes how to configure your Trino server to use :ref:`TLS
...
+
+    However, best practice is to instead use an external load balancer, as
+    discussed next.
```
Comment (hashhar) [inline docs/src/main/sphinx/security/tls.rst:75]:
```suggestion
    However, best practice is to terminate TLS at a load balancer instead as
    discussed next.
```

**Review by hashhar — APPROVED** [review]
LGTM since this is a lift-and-shift.

Left some comments since this could use much work - for someone unfamiliar with TLS configs there is some "not to the point" talk + some contradictions.

**Comment by hashhar** [comment]
@jhlodin Can you kindly create an issue so that we remember to follow up on this?

**Pull request merged**